### PR TITLE
[FLINK-26817] Update ingress docs with templating examples

### PR DESCRIPTION
**Pull request by morhidi**
- document ingress templates
  -  host based routing
  -  domain based routing

### CODE REVIEW

**Comment by morhidi** [comment]
cc @wangyang0918

File: docs/content/docs/operations/ingress.md [hunk 1]
```diff
@@ -25,33 +25,79 @@ under the License.
...
-  ingressDomain: flink.k8s.io
-  ...
+  ingress:
+    template: "flink.k8s.io/{{namespace}}/{{name}}(/|$)(.*)"
+    className: "nginx"
+    annotations:
+      nginx.ingress.kubernetes.io/rewrite-target: "/$2"
+```
+The `ingress` specification defines a mandatory `template` field and two optional fields `className` and `annotations`. 
+When the CR is submitted the Operator substitutes the template variables from metadata and creates an Ingress entry on the Kubernetes cluster.
```
Comment (wangyang0918) [inline docs/content/docs/operations/ingress.md:47]:
```suggestion
When the CR is submitted, the Operator substitutes the template variables from metadata and creates an Ingress entry on the Kubernetes cluster.  
```

File: docs/content/docs/operations/ingress.md [hunk 2]
```diff
@@ -25,33 +25,79 @@ under the License.
...
-  ...
+  ingress:
+    template: "flink.k8s.io/{{namespace}}/{{name}}(/|$)(.*)"
+    className: "nginx"
+    annotations:
+      nginx.ingress.kubernetes.io/rewrite-target: "/$2"
+```
+The `ingress` specification defines a mandatory `template` field and two optional fields `className` and `annotations`. 
+When the CR is submitted the Operator substitutes the template variables from metadata and creates an Ingress entry on the Kubernetes cluster.  
+Given the example above the Flink UI could be accessed at https://flink.k8s.io/default/advanced-ingress/ and the generated Ingress entry would be:
```
Comment (wangyang0918) [inline docs/content/docs/operations/ingress.md:48]: Could we add a note about that the trailing slash is necessary? I am afraid some users will run into this issue.

File: docs/content/docs/operations/ingress.md (hunk 1 above)
Comment (morhidi) [reply docs/content/docs/operations/ingress.md]: fixed

File: docs/content/docs/operations/ingress.md (hunk 2 above)
Comment (morhidi) [reply docs/content/docs/operations/ingress.md:48]: done

**Review by wangyang0918 — APPROVED** [review]
+1 for merging.

**Pull request merged**